### ISSUE TITLE
Failure to installing wasm-pack (intermittent behavior) on unneeded webpack type

### ISSUE THREAD

**Issue by adaptive** [cloudflare/wrangler#904]
### 🐛 Bug Report

On a `type = "webpack"` always installs wasm-pack (referenced previous #137 #849), it is not worrying, though the trouble is the download fails every other Push on CI/CD.

The issue might reside with the worker that manipulates `https://workers.cloudflare.com/get-binary/rustwasm/wasm-pack/v0.8.1/x86_64-unknown-linux-musl.tar.gz`

Fetch for AWS S3 and does not benefit from Cache API, my suggestion AWS S3 ➡️Worker KV (maybe expire key) ➡️Cache API.

At the moment it is always returning header `CF-Cache-Status: MISS`

#### Environment

GitHub Actions Ubuntu

#### Steps to reproduce

Run wrangler build

#### What did you expect to see?

```
Run wrangler build
⬇️ Installing wranglerjs...
npm WARN deprecated fsevents@1.2.9: One of your dependencies needs to upgrade to fsevents v2: 1) Proper nodejs v10+ support 2) No more fetching binaries from AWS, smaller package size
npm notice created a lockfile as package-lock.json. You should commit this file.
npm WARN wrangler-js@0.1.11 No description
npm WARN wrangler-js@0.1.11 No repository field.
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@1.2.9 (node_modules/fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.2.9: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})

added 323 packages from 198 contributors and audited 4225 packages in 7.66s
found 0 vulnerabilities

⬇️ Installing wasm-pack...
 Built successfully, built project size is 75 KiB.
```

#### What did you see instead?

```
Run wrangler build
⬇️ Installing wranglerjs...
npm WARN deprecated fsevents@1.2.9: One of your dependencies needs to upgrade to fsevents v2: 1) Proper nodejs v10+ support 2) No more fetching binaries from AWS, smaller package size
npm notice created a lockfile as package-lock.json. You should commit this file.
npm WARN wrangler-js@0.1.11 No description
npm WARN wrangler-js@0.1.11 No repository field.
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@1.2.9 (node_modules/fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.2.9: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})

added 323 packages from 198 contributors and audited 4225 packages in 11.351s
found 0 vulnerabilities

Error: could not download pre-built `wasm-pack` (failed to download from https://workers.cloudflare.com/get-binary/rustwasm/wasm-pack/v0.8.1/x86_64-unknown-linux-musl.tar.gz).
##[error]Process completed with exit code 1.
```

**Comment by xtuc** [comment]
Hi, thanks for the repo.

@EverlastingBugstopper is working on a fix

**Comment by lolcoolkat** [comment]
+1 for this. Builds could be sooooo much faster if it didn't install `wasm-pack` every time I published :P

**Comment by adaptive** [comment]
I am using wrangler in GitHub Action, with action/cache and "npx wrangler" the process speeds up a bit. Installing wranglerjs every and deploy is counterproductive in a CI/CD workflow.

**Comment by EverlastingBugstopper** [comment]
this issue was solved a while ago with #1344
